inference/huggingface/text-generation/inference-test.py

```diff
@@ -17,14 +17,31 @@
 parser.add_argument("--dtype", default="float16", type=str, choices=["float32", "float16", "int8"], help="data-type")
 parser.add_argument("--ds_inference", action='store_true', help="enable ds-inference")
 parser.add_argument("--use_kernel", action='store_true', help="enable kernel-injection")
+parser.add_argument("--replace_method", required=False, default='', type=str, help="replace method['', 'auto', 'dict']")
 parser.add_argument("--max_tokens", default=1024, type=int, help="maximum tokens used for the text-generation KV-cache")
 parser.add_argument("--max_new_tokens", default=50, type=int, help="maximum new tokens to generate")
 parser.add_argument("--greedy", action='store_true', help="greedy generation mode")
 parser.add_argument("--use_meta_tensor", action='store_true', help="use the meta tensors to initialize model")
 parser.add_argument("--use_cache", default=True, type=bool, help="use cache for generation")
+parser.add_argument("--test_performance", action='store_true', help="enable latency, bandwidth, and throughout testing")
 parser.add_argument("--local_rank", type=int, default=0, help="local rank")
 args = parser.parse_args()
 
+def print_perf_stats(latency_set, config, warmup=3):
+    # trim warmup queries
+    latency_set = list(latency_set)
+    latency_set = latency_set[warmup:]
+    count = len(latency_set)
+
+    if count > 0:
+        latency_set.sort()
+        avg = sum(latency_set) / count
+        num_layers = config.num_layers if hasattr(config,'num_layers') else config.num_hidden_layers
+        num_parameters = num_layers * config.hidden_size * config.hidden_size * 12
+        print("Avg Per Token Latency: {0:8.2f} ms".format(avg * 1000))
+        print("Avg BW: {0:8.2f} GB/s".format(1/avg * num_parameters * 2 / 1e9))
+        print("Avg flops: {0:8.2f} TFlops/s".format(1/avg * num_parameters * 2 / 1e12 * args.batch_size))
```

Comment:
Could you add parentheses to disambiguate the order of operations here?
For example:
`1/avg * ((num_parameters * 2) / 1e12) * args.batch_size`